nodejs/week2/src/backend/routes/meals.js

```diff
@@ -0,0 +1,54 @@
+//Bringing Dependencies
+const express = require('express');
+const app = express();
+const meals = require("../data/meals");
+const reviews = require("../data/reviews");
+
+//JSON response for with corresponding ID
+app.get('/meals/:id', (req,res)=>{
+    const id = Number(req.params.id);
+    const mealId = meals.filter(meal=>meal.id === id);
+    res.send(mealId);
+    if(mealId.length === 0){
+        res.status(401).send('Not found meal');
+}
+    else{
+        res.send(mealId);
+    }
+    
+});
+
+//JSON response for meals that has a price smaller than maxPrice
+app.get('/meals', (req,res)=>{
+    if(req.query.maxPrice){
+        const maxPrice = parseInt(req.params.maxPrice);
+        const mealSmallerThanMaxPrice = meals.filter((meal)=>meal.price <= maxPrice);
+        res.send(mealSmallerThanMaxPrice);
+    }
+//JSON response to Get meals that partially match a title.
+    else if(req.query.title){
+        const mealTitle = (req.query.title).toLowerCase();
+        const matchedMealTitle = meals.filter((meal)=>meal.title.toLowerCase().includes(mealTitle));
+        res.send(matchedMealTitle);
+    }
+//JSON response  to Get meals that has been created after the date
+    else if(req.query.createdAfter){
+        const createdDate = new Date(req.params.createdAfter);
+        const mealsCreatedAfter = meals.filter((meal)=>new Date(meal.createdAt) > createdDate);
+        res.send(mealsCreatedAfter);
+    }
+//JSON response to  get Only specific number of meals
+    else if(req.query.limit){
+        const limit = Number(req.params.limit);
+        const limitNumber = meals.slice(0, limit);
+        res.send(limitNumber);
+    }
+    else{
+        //JSON response for all meals
+        res.send(meals);
+
+        
+    }
+
```

Comment:
This can only handle one query param at a time right now. It should be able to handle all params and use them at once to filter the list of meals according to the params provided by the user.

In general, all of these query parameters are optional and could be provided or not with the request. If they are provided then you can use them to filter the meals according to `maxPrice`, `title`, etc. But if they are not provided you just don't need to use them. e.g.

`/meals` => give me all meals
`/meals?maxPrice=100` => give me all cheap meals under 100 crowns
`/meals?title=chicken` => give me all meals with chicken
`/meals?title=chicken&maxPrice=100` => give me all cheap meals with chicken

When the service behaves in this way in can be really useful because it provides a very rich functionality for the user.